frontend/common/stores/feature-list-store.ts

```diff
@@ -458,135 +465,216 @@ const controller = {
     mode: 'VALUE' | 'SEGMENT',
   ) => {
     store.saving()
-    API.trackEvent(Constants.events.EDIT_FEATURE)
-    const env: Environment = ProjectStore.getEnvironment(environmentId) as any
-    let environment_feature_versions = []
-    if (env.use_v2_feature_versioning) {
-      let featureStates
+    try {
+      API.trackEvent(Constants.events.EDIT_FEATURE)
+      const env: Environment = ProjectStore.getEnvironment(environmentId) as any
+      // Detect differences between change request and existing feature states
+      const res: { data: PagedResponse<FeatureState> } = await getFeatureStates(
+        getStore(),
+        {
+          environment: environmentFlag.environment,
+          feature: projectFlag.id,
+        },
+      )
+      let segments = null
       if (mode === 'SEGMENT') {
-        featureStates = segmentOverrides?.map((override: any, i: number) =>
-          convertSegmentOverrideToFeatureState(override, i, changeRequest),
-        )
-      } else {
-        featureStates = [
-          Object.assign({}, environmentFlag, {
-            enabled: flag.default_enabled,
-            feature_state_value: flag.initial_value,
-            live_from: flag.live_from,
-          }),
-        ]
+        const res = await getSegments(getStore(), {
+          include_feature_specific: true,
+          page_size: 1000,
+          projectId,
+        })
+        segments = res.data.results
       }
-
-      const { data: version } = await createAndSetFeatureVersion(getStore(), {
-        environmentId: env.id,
-        featureId: projectFlag.id,
-        featureStates,
-        liveFrom: changeRequest.live_from,
-        skipPublish: true,
+      const oldFeatureStates = res.data.results.filter((v) => {
+        return mode === 'VALUE' ? !v.feature_segment : !!v.feature_segment
       })
-      environment_feature_versions = [version.version_sha]
-    }
-    const prom = data
-      .get(
-        `${Project.api}environments/${environmentId}/featurestates/${environmentFlag.id}/`,
-      )
-      .then(() => {
-        const {
-          approvals,
-          featureStateId,
-          multivariate_options,
-          ...changeRequestData
-        } = changeRequest
 
-        const userApprovals = approvals.filter((u) => {
-          const keys = Object.keys(u)
-          return keys.includes('user')
-        })
+      let feature_states_to_create:
+          | Req['createFeatureVersion']['feature_states_to_create']
+          | undefined = undefined,
```

Comment:
I don't quite understand what's happening here?

Reply:
feature_states_to_create etc are assigned a value if versioning is enabled, otherwise it doesn't send anything (undefined)

Reply:
gotcha, sorry, I understand now 👍